packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -281,143 +282,35 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
-                            }
-                        }
-                    }
+            const hostPolicy: ISnapshotOptions = {
+                deltas: 1,
+                channels: 1,
+                blobs: 2,
+                ...this.hostPolicy.snapshotOptions,
+            };
 
-                    // Populate the cache with paths from id-to-path mapping.
-                    for (const blob of this.blobCache.values()) {
-                        const path = blobsIdToPathMap.get(blob.id);
-                        // If this is the first container that was created for the service, it cannot be
-                        // the summarizing container (because the summarizing container is always created
-                        // after the main container). In this case, we do not need to do any hashing
-                        if (path) {
-                            // Schedule the hashes for later, but keep track of the tasks
-                            // to ensure they finish before they might be used
-                            const hashP = hashFile(IsoBuffer.from(blob.content, blob.encoding)).then((hash: string) => {
-                                this.blobsShaToPathCache.set(hash, path);
-                            });
-                            this.blobsCachePendingHashes.add(hashP);
-                            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-                            hashP.finally(() => {
-                                this.blobsCachePendingHashes.delete(hashP);
-                            });
-                        }
-                    }
-                }
+            // No limit on size of snapshot, as otherwise we fail all clients to summarize
+            if (this.hostPolicy.summarizerClient) {
+                hostPolicy.mds = undefined;
+            }
+            try {
+                return getWithRetryForTokenRefresh(async (tokenFetchOptions) => {
+                    this.isTreesLatestSecondCallEvent(tokenFetchOptions);
 
-                this.ops = ops;
-                return id ? [{ id, treeId: undefined! }] : [];
-            });
+                    // Use the POST call for TreesLatest so as to save the OPTIONS call.
+                    const cachedSnapshot = await this.fetchSnapshotWithMultiPartRequest(tokenFetchOptions, hostPolicy);
+                    return this.processSnapshotAndGetVersion(cachedSnapshot);
+                });
+            } catch (error) {
+                this.logger.sendErrorEvent({ eventName: "TreeLatest_FallBackToNonMultiPartRequest" });
+                // If the TreesLatest fails using the POST call, then fallback to the original get version of TreesLatest call.
+                return getWithRetryForTokenRefresh(async (tokenFetchOptions) => {
+                    this.isTreesLatestSecondCallEvent(tokenFetchOptions);
```

Comment:
I assume this is the telemetry we will use to determine how effective the switch over is?

Reply:
This telemetry event if for refreshing the token if the original token is expired.

Reply:
This is the one for post call failure and when callback code is called.:
this.logger.sendErrorEvent({ eventName: "TreeLatest_FallBackToNonMultiPartRequest" });